### PR TITLE
Scheduler: wrap errors for framework/runtime

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Wrap all errors under framework/runtime.

**Which issue(s) this PR fixes**:

Part of #94695

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/runtime/framework.go [hunk 1]
```diff
@@ -538,7 +538,7 @@ func (f *frameworkImpl) RunFilterPlugins(
 			if !pluginStatus.IsUnschedulable() {
 				// Filter plugins are not supposed to return any status other than
 				// Success or Unschedulable.
-				errStatus := framework.NewStatus(framework.Error, fmt.Sprintf("running %q filter plugin for pod %q: %v", pl.Name(), pod.Name, pluginStatus.Message()))
+				errStatus := framework.AsStatus(fmt.Errorf("running %q filter plugin for pod %q: %w", pl.Name(), pod.Name, pluginStatus.AsError()))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/runtime/framework.go:541]: can you trace what happens with the error in higher function calls?

Maybe the name of the Pod is added again at some point.

Comment (gavinfish) [reply pkg/scheduler/framework/runtime/framework.go:541]: no, we use the function in both schedule and preempt and they are all clean.

Comment (gavinfish) [reply pkg/scheduler/framework/runtime/framework.go:541]: Get your point now. Nice catch!

**Comment by alculquicondor** [comment]
please squash

**Comment by gavinfish** [comment]
squashed

**Pull request merged**